UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs

```diff
@@ -31,119 +33,124 @@ public async Task SetupAsync()
+                                
+                FileContentHelper.CheckMediaTxtFileContent(Path.Combine(folderPath, Config.POSConfig.LargeExchangeSetMediaFileName), mediaNumber);
+                mediaNumber++;
             }
         }
 
         [Test]
         [Category("SmokeTest")]
-        public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnINFOFolderIsGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnINFOFolderIsGenerated()
         {
-            for (int mediaNumber = 1; mediaNumber <= 2; mediaNumber++)
+            int mediaNumber = 1;
+            foreach (string folderPath in DownloadedFolderPath)
             {
-                var FolderName = $"M0{mediaNumber}X02";
-                DownloadedFolderPath = await FileContentHelper.ExchangeSetLargeFile(ApiEssResponse, FssJwtToken, FolderName);
-                bool checkFolder = FssBatchHelper.CheckforFolderExist(DownloadedFolderPath, Config.POSConfig.LargeExchangeSetInfoFolderName);
-                
+                bool checkFolder = FssBatchHelper.CheckforFolderExist(folderPath, Config.POSConfig.LargeExchangeSetInfoFolderName);
                 Assert.IsTrue(checkFolder, $"Folder not Exist in the specified folder path :");
+                
+                mediaNumber++;
             }
         }
 
         [Test]
         [Category("SmokeTest")]
-        public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnADCFolderIsGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnADCFolderIsGenerated()
         {
-            for (int mediaNumber = 1; mediaNumber <= 2; mediaNumber++)
+            int mediaNumber = 1;
+            foreach (string folderPath in DownloadedFolderPath)
             {
-                var FolderName = $"M0{mediaNumber}X02";
-                DownloadedFolderPath = await FileContentHelper.ExchangeSetLargeFile(ApiEssResponse, FssJwtToken, FolderName);
-                bool checkFolder = FssBatchHelper.CheckforFolderExist(Path.Combine(DownloadedFolderPath, Config.POSConfig.LargeExchangeSetInfoFolderName), Config.POSConfig.LargeExchangeSetAdcFolderName);
-                
+                bool checkFolder = FssBatchHelper.CheckforFolderExist(Path.Combine(folderPath, Config.POSConfig.LargeExchangeSetInfoFolderName), Config.POSConfig.LargeExchangeSetAdcFolderName);
                 Assert.IsTrue(checkFolder, $"Folder not Exist in the specified folder path :");
+
+                mediaNumber++;
             }
         }
 
         [Test]
         [Category("SmokeTest")]
-        public async Task WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAReadMeTxtFileIsGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAReadMeTxtFileIsGenerated()
         {
+            int mediaNumber = 1;
```

Comment:
I don't think mediaNumber is used in this function, other than to increment. Could you confirm and remove?

Reply:
Updated.

Thanks